src/styles/_catalog.scss

```
@@ -1,22 +1,42 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-template-columns: repeat(auto-fill, 200px);
+  gap: 48px;
+  justify-content: center;
 
   max-width: 944px;
   margin: 0 auto;
 
   &__card {
-    display: flex;
+    display: grid;
+    box-sizing: border-box;
     justify-content: center;
     align-items: center;
     width: 200px;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
   &__card:not(:nth-child(4n)) {
     margin-right: 48px;
   }
 }
+
+@media (max-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(3, 200px);
+  }
+}
+
+@media (max-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(2, 200px);
+  }
+}
+
+@media (max-width: 488px) {
+  .catalog {
+    grid-template-columns: repeat(1, 200px);
+  }
+}
```

```suggestion
@media (max-width: 488px) {
  .catalog {
    grid-template-columns: repeat(1, 200px);
  }
}

@media (max-width: 768px) {
  .catalog {
    grid-template-columns: repeat(2, 200px);
  }
}

@media (max-width: 1024px) {
  .catalog {
    grid-template-columns: repeat(3, 200px);
  }
}
```
Распологаай медиа запросы по мере возрастания

---

Потому что в таком порядке нижний медиа запрос будет перекрывать верхний на ширине до 1024px